### ISSUE TITLE
Quality Selection

### ISSUE THREAD

**Issue by qome** [TheUnlocked/Better-Mixer#56]
It would be nice to have an option to select a pre-defined quality when loading a stream. As it stands now Mixer serves up the highest quality available, which is pretty wasteful. I don't need 1440p video in a like 900p sized player. As a matter of fact most of the time I prefer a pretty low resolution.

**Comment by TheUnlocked** [comment]
I agree and I've looked into this multiple times. Unfortunately Mixer is extremely stubborn on letting third-party code touch quality options so I haven't been able to do anything about this, and I may not be able to unless something about how the Mixer player works changes.

**Comment by qome** [comment]
I've attempted to use the developer tools to limit the tab's bandwidth to extremely low values, but that seems to have no impact on the player. Any ideas from that angle?

**Comment by TheUnlocked** [comment]
Developer tools only work while the developer window is open, so that isn't possible.

Earlier today I looked into how Mixer changes resolution some more, and while I have figured out a potential system for changing the stream resolution, I've found no way to actually determine the available quality options for a particular stream and build an appropriate object payload based on them. From what I can tell so far, all of the APIs to do that are contained within closures that I can't access from the outside (the seemingly low-hanging fruit like `MixerPlayer.selectQualityByName` and `MixerPlayer.selectQualityByBitrate` all consume objects or functions that I don't have access to, and don't actually solve the issue of not having the list of available transcoding options).

There is a sort of nuclear option, which is to fetch transcoding information from the `/channels/{channelId}/manifest.m3u8` endpoint, parse it, and attempt to build our own imitation payload object from the partial information provided by the manifest. However, that could end up being a nightmare to implement, and even more of a nightmare to verify whether it works and will continue to work.

Unfortunately the easier `/transcodes` endpoint seems like it either doesn't have any real support anymore or is so lacking on documentation that I wouldn't even know where to begin.
